pkg/adsc/adsc.go

```diff
@@ -391,58 +521,31 @@ func (a *ADSC) handleRecv() {
 				}
 
 			default:
-				if len(gvk) != 3 {
-					continue
-				}
-				// Generic - fill up the store
-				if a.Store != nil {
-					m := &mcp.Resource{}
-					err = types.UnmarshalAny(&types.Any{
-						TypeUrl: rsc.TypeUrl,
-						Value:   rsc.Value,
-					}, m)
-					if err != nil {
-						continue
-					}
-					val, err := mcpToPilot(m)
-					if err != nil {
-						continue
-					}
-					val.GroupVersionKind = resource.GroupVersionKind{gvk[0], gvk[1], gvk[2]}
-					if err != nil {
-						adscLog.Warna("Invalid data ", err, " ", string(valBytes))
-					} else {
-						cfg := a.Store.Get(val.GroupVersionKind, val.Name, val.Namespace)
-						if cfg == nil {
-							_, err = a.Store.Create(*val)
-							if err != nil {
-								continue
-							}
-						} else {
-							_, err = a.Store.Update(*val)
-							if err != nil {
-								continue
-							}
-						}
-					}
-					if a.LocalCacheDir != "" {
-						strResponse, err := json.MarshalIndent(val, "  ", "  ")
-						if err != nil {
-							continue
-						}
-						err = ioutil.WriteFile(a.LocalCacheDir+"_res."+
-							val.GroupVersionKind.Kind+"."+val.Namespace+"."+val.Name+".json", strResponse, 0644)
-						if err != nil {
-							continue
-						}
-					}
-				}
+				_ = a.handleMCP(gvk, rsc, valBytes)
```

Comment:
if we are going to use adsc in our real code we probably want to stop doing things like ignoring errors?